superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterSets/FilterSets.tsx

```diff
@@ -133,21 +144,59 @@ const FilterSets: React.FC<FilterSetsProps> = ({
     );
   };
 
+  const handleInvalidate = (id: string) => {
+    const filtersSet = filterSets[id];
+    // We need remove invalid filters from filter set
+    const newFilters = Object.values(filtersSet?.dataMask?.nativeFilters ?? [])
+      .filter(dataMask => {
+        const { id } = dataMask as MaskWithId;
+        // if we have extra filters in filter set don't add them to selected data mask || if we have filters with changed metadata not apply them
+        return !(
+          !filterValues.find(filter => filter?.id === id) ||
+          !areObjectsEqual(filters[id], filtersSet?.nativeFilters?.[id])
+        );
```

Comment:
Could we move this to a helper function to avoid duplication (same logic used on lines 133-136).